sympy/logic/boolalg.py

```diff
@@ -1637,6 +1808,84 @@ def POSform(variables, minterms, dontcares=None):
     return And(*[_convert_to_varsPOS(x, variables) for x in essential])
 
 
+def ANF(variables, truthvalues):
+    """
+    The ANF function converts the list of truth values to
+    Algebraic Normal Form (ANF).
+
+    The variables must be given as the first argument.
+
+    Return True, False, logical And function (i.e., the
+    "Zhegalkin monomial") or logical Xor function (i.e.,
+    the "Zhegalkin polynomial").
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import ANF
+    >>> ANF(['x'], [1, 0])
+    Xor(x, True)
+    >>> ANF(['x', 'y'], [0, 1, 1, 1])
+    Xor(x, y, And(x, y))
+
+    References
+    ==========
+
+    .. [2] en.wikipedia.org/wiki/Zhegalkin_polynomial
+
+    """
+    variables = [sympify(v) for v in variables]
+
+    cfs = coeffs(truthvalues)
+    terms = []
+
+    for i, t in enumerate(product([0, 1], repeat=len(variables))):
+        if cfs[i] == 1:
+            terms.append(t)
+
+    return Xor(*[_convert_to_varsANF(x, variables) for x in terms])
+
+
+def coeffs(truthvalues):
+    """
+    Convert a list of truth values of some boolean expression
+    to the list of coefficients of the polynomial mod 2 (exclusive
+    disjunction) representing the boolean expression in ANF
+    (i.e., the "Zhegalkin polynomial").
+
+    There are 2^n possible Zhegalkin monomials in n variables, since
+    each monomial is fully specified by the presence or absence of
+    each variable.
+
+    A given monomial's presence or absence in a polynomial corresponds
+    to that monomial's coefficient being 1 or 0 respectively.
+
+    Examples
+    ========
+    >>> from sympy.logic.boolalg import coeffs
+    >>> truthvalues = [0, 1, 1, 0, 0, 1, 0, 1]
+    >>> coeffs(truthvalues)
+    [0, 1, 1, 0, 0, 0, 1, 0]
```

Comment:
In order to represent the Boolean expression in Algebraic Normal Form (ANF) it is enough to know the list of indexes of monomials with non zero coefficients.

For example, the list <code>[1, 2, 6]</code> can be considered as the Boolean expression with three variables in ANF.  In this case, ANF contains only 1-st, 2-nd and 6-th monomials. We can rewrite the list like this <code>[0, 1, 1, 0, 0, 0, 1, 0]</code>.

Monomial can be deduced from its index. We need to convert the index of monomial from decimal to binary. For example, <code>[1, 2, 6] -> [[0, 0, 1], [0, 1, 0], [1, 1, 0]] </code>. If our variables <code>[a, b, c]</code>, then monomials <code>a<sup>0</sup>b<sup>0</sup>c<sup>1</sup> = c </code>; <code>a<sup>0</sup>b<sup>1</sup>c<sup>0</sup> = b </code>; <code>a<sup>1</sup>b<sup>1</sup>c<sup>0</sup> = ab </code>. ANF in this example: <code>Xor(b, c, And(a, b))</code>.

Converting truth values to ANF is similar to converting minterms to SOP form, or converting maxterms to POS form.
In order to convert minterms to SOP form, we only need to know the indexes of the minterms. In order to convert maxterms to POS form, we only need to know the indexes of the maxterms.

Converting minterms to SOP form:
<code> [1, 2, 4, 7] -> [[0, 0, 1], [0, 1, 0], [1, 0, 0], [1, 1, 1]] -> (~a & ~b & c) | (~a & b & ~c) | (a & ~b & ~c) | (a & b & c) </code>